concept/type/impl/RoleTypeImpl.java

```diff
@@ -129,7 +135,9 @@ void validateDelete() {
     }
 
     private FunctionalIterator<RoleImpl> getInstances() {
-        return instances(RoleImpl::of);
+        return getSubtypes().filter(t -> !t.isAbstract())
+                .flatMap(t -> graphMgr.data().getReadable(t.vertex))
+                .map(RoleImpl::of);
```

Comment:
this could be seekable but we only use it for delete, so no point introducing comparable overhead right now